drivers/flash/spi_nor.c

```diff
@@ -213,7 +214,11 @@ static int spi_nor_read(struct device *dev, off_t addr, void *dest,
 err:
```

Comment:
Why not return the error code returned by `spi_nor_cmd_addr_read` and adding a `SYNC_UNLOCK()`?